add homelessness to to_s methods in address and residence models

colinxfleming opened the pull request:
Sorry this took so long! I spent a lot of time trying to get this to run locally to confirm the changes were working right. 

This commit alters methods in the residence and address models so that an address marked as 'Homeless' shows up as such in the view. The display logic is in app/models/residence.rb, starting line 35.

Added a fixture and a test to go with it as well.

This should fix #311 . I couldn't recreate the 'doesn't save' issue either so I'm hoping this passes muster. Let me know if I can do anything else on this or if you'd prefer it implemented a different way.

jrunningen commented on test/models/residence_test.rb:
```diff
@@ -10,4 +10,26 @@ def test_valid
+    assert_match /homeless/i, residence.to_s
+
+    # tests 'homeless'
+    residence.current = true
```
Isn't residence.current already true at this point?

colinxfleming replied:
Line 22, right? Whoops, sorry. 

Changing now.

jrunningen commented on test/models/residence_test.rb:
```diff
@@ -10,4 +10,26 @@ def test_valid
+
+    # tests 'currently homeless'
+    residence = Residence.new address_id: addr.id, current: true
+    assert_match /homeless/i, residence.to_s
```
assert_match /currently homeless/i

jrunningen replied:
Actually, for these test cases, it makes sense to just test the whole string instead of doing regex comparisons.  That way, you verify that "current" is present in one case and absent in another.  Do assert_equal "Currently homeless", and so on for the other assertions.

colinxfleming replied:
Sorry about the delay getting these back -- this past week turned into a doozy for me. 

Changed these to all assert_equals from matches.

jrunningen commented on app/models/residence.rb:
```diff
@@ -33,8 +33,12 @@ def initialize_residence
+     "Currently homeless"
+    elsif current
       "Current residence at #{address}"
+    elsif /homeless/i.match(address.to_s)
```
You use two different regexes to indicate homeless: /^homeless/i and /homeless/i.  Consider defining a method "homeless?", and calling that instead, to keep the code dry.

jrunningen replied:
The "homeless?" Method probably belongs in the Address model, since you have a regex there too.

colinxfleming replied:
Good call on this. Plopped a homeless? method in the address model, which returns true if the street given has homeless in the name, and called that in the residence model instead.

colinxfleming commented:
Fixed up the recommendations per @jrunningen 's comments -- edits to the asserts in the residence test and drying up the homelessness check by tossing it into the address model.

Also added a test for the homeless? method in the address model tests, just to be on the safe side. 

Again, sorry for the delay getting these in. LMK if you'd like further changes b/w thanks for your patience, haha.

jrunningen commented:
Awesome work!  Thanks for your persistence.  Keep it up.

jrunningen commented:
This change is now active on the districthousing.org demo site.

colinxfleming commented:
I'm here to hustle! $$$
